projects/project-2/assignment/Assginment 2.md

```diff
@@ -0,0 +1,241 @@
+    ?property owl:inverseOf ?inverseProperty .
+    
+    # Check if the inverse property is the same as the symmetric property itself or also symmetric
+    FILTER (?property = ?inverseProperty || EXISTS { ?inverseProperty a owl:SymmetricProperty })
+}
+    
+  </code>
+</pre>
+
+6. Detect Classes Without a Subclass (Leaf Classes). This query detects classes that do not have any subclasses.
+   These "leaf" classes might be indicators of incomplete modeling if too many are present. (level 3 - 20 pts)
+
+<pre>
+  <code>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+
+SELECT ?class
+WHERE {
+    ?class a rdfs:Class .
+    
+    # Find classes that do not have any subclasses
+    FILTER NOT EXISTS {
+        ?subClass rdfs:subClassOf ?class .
+    }
+} 
+  </code>
+</pre>
+
+
+
+7. Detect Properties with Conflicting Domain and Range: This query checks for properties that are used with instances whose types conflict with the
+   property's declared domain and range, which could result in incorrect or inconsistent usage of the properties. (level 3 20 pts)
+
+<pre>
+  <code>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+
+SELECT ?property ?subject ?object
+WHERE {
+    ?property a rdf:Property ;
+              rdfs:domain ?domain ;
+              rdfs:range ?range .
+    
+    ?subject rdf:type ?subjectClass .
+    ?subject ?property ?object .
+    ?object rdf:type ?objectClass .
+
+    # Check if the subject and object are not conforming to the domain and range
+    FILTER(?subjectClass != ?domain || ?objectClass != ?range)
+}
+
+  </code>
+</pre>
+
+8. Detect Classes without Subclasses or Instances: This query identifies classes that neither have subclasses nor instances,
+   which might indicate unused or improperly modeled classes. (level 2 - 25 pts)
+
+<pre>
+  <code>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+
+SELECT ?class
+WHERE {
+    ?class a rdfs:Class .
+    
+    # Check for missing subclasses
+    FILTER NOT EXISTS {
+        ?subclass rdfs:subClassOf ?class .
+    }
+    
+    # Check for missing instances
+    FILTER NOT EXISTS {
+        ?instance rdf:type ?class .
+    }
+}
+  </code>
+</pre>
+    
+
+
+9. Detect Classes Without `owl:disjointWith` Declarations: This query checks for classes that do not have an `owl:disjointWith` declaration,
+    which helps ensure logical separation between classes. (level 8)
+
+<pre>
+  <code>
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+
+SELECT ?classA
+WHERE {
+    ?classA a rdfs:Class .
+    
+    # Check if the class lacks an owl:disjointWith declaration
+    FILTER NOT EXISTS {
+        ?classA owl:disjointWith ?classB .
+    }
+}
+
+  </code>
+</pre>
```

Comment:
Looking at the assignment I think is missing the severity level for each query ("Warning" or "Error")

Reply:
Ohh right. I will add them too! Thanks!